test/terra/extensions/test_snapshot_all.py

```diff
@@ -0,0 +1,54 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+
+import unittest
+
+from qiskit import QuantumCircuit, assemble
+from qiskit.extensions.exceptions import ExtensionError
+from qiskit.providers.aer import QasmSimulator
+from qiskit.providers.aer import UnitarySimulator
+from qiskit.providers.aer.extensions import Snapshot
+
+from qiskit.providers.aer.extensions.snapshot_density_matrix import SnapshotDensityMatrix
+from qiskit.providers.aer.extensions.snapshot_expectation_value import SnapshotExpectationValue
+from qiskit.providers.aer.extensions.snapshot_probabilities import SnapshotProbabilities
+import qiskit.quantum_info as qi
+
+class TestSnapshotsExtension(unittest.TestCase):
```

Comment:
The `test.terra.extensions` directory is for tests of the snapshot instruction objects themselves (no execution).

There are already density matrix snapshot tests in `test/terra/backends/qasm_simulator/qasm_snapshot.QasmSnapshotDensityMatrixTests`. Any additional tests should be added there so they are run automatically on any qasm method that supports it (and those tests might need to be updated for these changes).

The unitary snapshot test should be added to the `test_unitary_simulator` tests.

Reply:
Addressed by 53a8b19